### PR TITLE
[final] Subscriber attributes

### PR DESCRIPTION
**Pull request by aboedo**
This adds methods to access subscriber attributes, as well as invalidate purchaser info cache. 
It also bumps both `purchases-ios` and `purchases-hybrid-common` to the latest versions. 

I tested all methods on the sample app, and tried getting the push token through the [cordova-plugin](https://medium.com/@anjana.pai.rathod/how-to-send-push-notifications-to-ios-using-phonegap-or-cordova-and-java-8ebf95ccdd9e), got a string and it was sent to the backend correctly. 

### Requirements: 
- [X] ~[purchases-hybrid-common PR]~(https://github.com/RevenueCat/purchases-hybrid-common/pull/5)

### CODE REVIEW

File: src/android/PurchasesPlugin.java [hunk 1]
```diff
@@ -179,6 +181,44 @@ private void isAnonymous(JSONArray productIDs, CallbackContext callbackContext)
...
+    }
+
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "setDisplayName")
+    private void setDisplayName(String displayName, CallbackContext callbackContext) {
+        CommonKt.setDisplayName(displayName);
+    }
+
+    @PluginAction(thread = ExecutionThread.WORKER, actionName = "setPushToken")
+    private void setPushToken(String pushToken, CallbackContext callbackContext) {
+        CommonKt.setPushToken(pushToken);
```
Comment (vegaro) [inline src/android/PurchasesPlugin.java]: This is interesting. The `callbackContext` is not notified by doing `callbackContext.success()`. But if you check the `setDebugLogsEnabled` we are not doing it either. If you check the iOS code, we do notify tho....We should check what's the right thing to do

File: src/plugin/plugin.ts [hunk 2]
```diff
@@ -878,6 +878,102 @@ class Purchases {
...
+      [attributes]
+    );
+  }
+  
+  /**
+   * Subscriber attribute associated with the email address for the user
+   *
+   * @param email Empty String or null will delete the subscriber attribute.
+   */
+  public static setEmail(email: string) {
```
Comment (vegaro) [inline src/plugin/plugin.ts]: these should be `string | null`

Comment (aboedo) [reply src/plugin/plugin.ts:921]: updating

File: package.json [hunk 3]
```diff
@@ -21,8 +21,8 @@
     "cordova-android"
   ],
   "scripts": {
-    "fetch:ios:sdk": "./scripts/download-purchases-framework.sh 3.0.2",
-    "fetch:ios:common": "./scripts/download-purchases-common.sh 1.0.5",
+    "fetch:ios:sdk": "./scripts/download-purchases-framework.sh 3.1.2",
+    "fetch:ios:common": "./scripts/download-purchases-common.sh 1.0.8",
```
[the remark this answers is not in the document]
Comment (aboedo) [reply package.json]: updated

File: plugin.xml [hunk 4]
```diff
@@ -16,7 +16,7 @@
                 <param name="android-package" value="com.revenuecat.purchases.PurchasesPlugin" />
             </feature>
         </config-file>
-        <framework src="com.revenuecat.purchases:purchases:3.0.5" />
+        <framework src="com.revenuecat.purchases:purchases:3.1.2" />
```
[the remark this answers is not in the document]
Comment (aboedo) [reply plugin.xml]: fixed

File: src/android/PurchasesPlugin.java (hunk 1 above)
Comment (aboedo) [reply src/android/PurchasesPlugin.java:216]: I can't find good docs on it. It seems cleaner to call success, though. I'll add it

File: VERSIONS.md [hunk 5]
```diff
@@ -1,5 +1,7 @@
 | Version | iOS version | Android version | Common files version |
 | ------- | ----------- | --------------- | -------------------- |
+| 1.0.6   | 3.2.2       | 3.1.0           | 1.0.11               |
```
Comment (vegaro) [inline VERSIONS.md:3]: 1.1.0?

Comment (vegaro) [inline VERSIONS.md:3]: 1.1.0?

Comment (aboedo) [reply VERSIONS.md:3]: sounds good, updating

**Pull request merged**